### ISSUE TITLE
Curse of Opulence bug

### ISSUE THREAD

**Issue by jeffwadsworth** [magefree/mage#5566]
https://www.slightlymagic.net/forum/viewtopic.php?f=70&t=23164&p=229567#p229567
http://gatherer.wizards.com/Pages/Card/ ... eid=433265

**Comment by jeffwadsworth** [comment]
The trigger on this card will happen if either a player or a planeswalker is attacked.  This is incorrect.

**Comment by JayDi85** [comment]
You can add `ignorePlaneswalkerAttack` param to EnchantedPlayerAttackedTriggeredAbility (default is false). Same param for `getPlayerDefenders` (default false). Calls it from EnchantedPlayerAttackedTriggeredAbility... And change `getPlayerDefenders` to use new param.

**Comment by JayDi85** [comment]
`Curse of Shallow Graves` and other cards (search by `getPlayerDefenders(game)`) have same condition and code and needs to fixes too (same calls with extra param -- see above).

**Comment by jeffwadsworth** [comment]
@JayDi85 Yes, I tried things like that but the code for combat is so convoluted,  I dove into trying to straighten that out.  A nightmare.  It just made me move on.  Perhaps you can take a fresh take at it.

**Comment by JayDi85** [comment]
Fixed via e285b2770d3f36bd23852812e312cf286094654a

**Comment by JayDi85** [comment]
Need to checks other "curse of" cards and replace `checkTrigger` code to use standard `getPlayerDefenders` call like other working curse cards (see e285b2770d3f36bd23852812e312cf286094654a).

From forum:
> Curse of Predation
> Need to checks other "curse of" cards with custom code and replace with correct (
> Enchant player
> Whenever a creature attacks enchanted player, put a +1/+1 counter on it.
>
> I had a Nissa, Voice of Zendikar under my control and my opponent had a Curse of Predation targetting me. When they attacked Nissa with their Bloodbraid Elf, the Curse of Predation triggered on it. I don't think this is correct, and there is an oracle ruling explaining this:
> "The ability won’t trigger when a creature attacks a planeswalker controlled by the enchanted player."

**Comment by jeffwadsworth** [comment]
@JayDi85 Are you doing this or does it need to be done?

**Comment by JayDi85** [comment]
Yep, I'll fix it.

**Comment by Zerrisx** [comment]
So... did you fix it?  Or does it still need to be done?

**Comment by JayDi85** [comment]
It's still actual. Cards to check: `final class CurseOf`
